File: Bars.GkhCR/libs/B4/model/Dict/ProgramCR.js
Ext.define('B4.model.dict.ProgramCr', {
    extend: 'B4.base.Model',
    idProperty: 'Id',
    proxy: {
        type: 'b4proxy',
        controllerName: 'ProgramCr'
    },
    fields: [
        { name: 'Id', useNull: true },
        { name: 'Name'}, 
        { name: 'Description'},
        { name: 'Code' },
        { name: 'UsedInExport', defaultValue: false },
        { name: 'NotAddHome', defaultValue: false },
        { name: 'MatchFl', defaultValue: false },
        { name: 'ForSpecialAccount', defaultValue: false },
        { name: 'UseForReformaAndGisGkhReports', defaultValue: false },
        { name: 'FinanceSource' },
        { name: 'Contragent' },
        { name: 'ContragentName' },
        { name: 'Period' },
        { name: 'PeriodName' },
        { name: 'IsCreateByDpkr', defaultValue: false },
        { name: 'TypeVisibilityProgramCr', defaultValue: 10 },
        { name: 'TypeProgramCr', defaultValue: 10 },
        { name: 'TypeProgramStateCr', defaultValue: 10 },
        { name: 'AddWorkFromLongProgram', defaultValue: 0 },
        { name: 'NormativeDoc' },
        { name: 'File', defaultValue: null },
        { name: 'DocumentNumber' },
        { name: 'DocumentDate' },
        { name: 'DocumentDepartment' },
        { name: 'ImportContract', defaultValue: false },
        { name: 'GovCustomer' }
    ]
});

File: Bars.GkhCR/Entities/Dicts/ProgramCR.cs
namespace Bars.GkhCr.Entities
{
    using Bars.B4.Modules.FileStorage;
    using System;

    using Bars.Gkh.Entities;
    using Bars.Gkh.Entities.Dicts;
    using Bars.Gkh.FormatDataExport.ExportableEntities;
    using Bars.GkhCr.Enums;

    /// <summary>
    /// Программа капитального ремонта
    /// </summary>
    public class ProgramCr : BaseGkhEntity, IHaveExportId
    {
        /// <summary>
        /// Видимость
        /// </summary>
        public virtual TypeVisibilityProgramCr TypeVisibilityProgramCr { get; set; }

        /// <summary>
        /// Период
        /// </summary>
        public virtual Period Period { get; set; }

        /// <summary>
        /// Тип КР
        /// </summary>
        public virtual TypeProgramCr TypeProgramCr { get; set; }

        /// <summary>
        /// Состояние
        /// </summary>
        public virtual TypeProgramStateCr TypeProgramStateCr { get; set; }

        /// <summary>
        /// Код
        /// </summary>
        public virtual string Code { get; set; }

        /// <summary>
        /// Используется при экспорте
        /// </summary>
        public virtual bool UsedInExport { get; set; }

        /// <summary>
        /// Не доступно добавление домов
        /// </summary>
        public virtual bool NotAddHome { get; set; }

        /// <summary>
        /// Примечание
        /// </summary>
        public virtual string Description { get; set; }

        /// <summary>
        /// Наименование программы
        /// </summary>
        public virtual string Name { get; set; }

        /// <summary>
        /// Соответствует ФЗ
        /// </summary>
        public virtual bool MatchFl { get; set; }

        /// <summary>
        /// Для специальных счетов
        /// </summary>
        public virtual bool ForSpecialAccount { get; set; }

        /// <summary>
        /// Использовать для отчетов Реформы и ГИС ЖКХ
        /// </summary>
        public virtual bool UseForReformaAndGisGkhReports { get; set; }

        /// <summary>
        /// Добавление видов работ из ДПКР
        /// </summary>
        public virtual AddWorkFromLongProgram AddWorkFromLongProgram { get; set; }

        /// <summary>
        /// Постановление об утверждении КП
        /// </summary>
        public virtual NormativeDoc NormativeDoc { get; set; }

        /// <summary>
        /// Файл
        /// </summary>
        public virtual FileInfo File { get; set; }
        
        /// <summary>
        /// Номер документа
        /// </summary>
        public virtual string DocumentNumber { get; set; }

        /// <summary>
        /// Дата документа
        /// </summary>
        public virtual DateTime? DocumentDate { get; set; }

        /// <summary>
        /// Орган, принявший документ
        /// </summary>
        public virtual string DocumentDepartment { get; set; }

        /// <summary>
        /// ГИС ЖКХ GUID
        /// </summary>
        public virtual string GisGkhGuid { get; set; }

        /// <summary>
        /// ГИС ЖКХ Transport GUID
        /// </summary>
        public virtual string GisGkhTransportGuid { get; set; }

        /// <summary>
        /// ГИС ЖКХ GUID вложения документа плана
        /// </summary>
        public virtual string GisGkhDocumentAttachmentGuid { get; set; }

        /// <summary>
        /// ГИС ЖКХ GUID документа плана
        /// </summary>
        public virtual string GisGkhDocumentGuid { get; set; }

        /// <summary>
        /// ГИС ЖКХ Transport GUID документа плана
        /// </summary>
        public virtual string GisGkhDocumentTransportGuid { get; set; }

        /// <summary>
        /// Флажок для выгрузки договоров
        /// </summary>
        public virtual bool ImportContract { get; set; }
        
        /// Государственный заказчик
        /// </summary>
        public virtual Contragent GovCustomer { get; set; }

        /// <inheritdoc />
        public virtual long ExportId { get; set; }
    }
}
